app/javascript/packages/spinner-button/spinner-button-element.spec.ts

```diff
@@ -14,20 +13,28 @@ describe('SpinnerButtonElement', () => {
 
   interface WrapperOptions {
     actionMessage?: string;
-
     tagName?: string;
-
     spinOnClick?: boolean;
+    inForm?: boolean;
+    isButtonTo?: boolean;
   }
 
-  function createWrapper({ actionMessage, tagName = 'a', spinOnClick }: WrapperOptions = {}) {
-    document.body.innerHTML = `
+  function createWrapper({
+    actionMessage,
+    tagName = 'a',
+    spinOnClick,
+    inForm,
+    isButtonTo,
+  }: WrapperOptions = {}) {
+    let html = `
       <lg-spinner-button
         long-wait-duration-ms="${longWaitDurationMs}"
         ${spinOnClick === undefined ? '' : `spin-on-click="${spinOnClick}"`}
       >
         <div class="spinner-button__content">
+          ${isButtonTo ? '<form action="#">' : ''}
           ${tagName === 'a' ? '<a href="#">Click Me</a>' : '<input type="submit" value="Click Me">'}
+          ${isButtonTo ? '</form>' : ''}
```

Comment:
is this manual closing tag an opportunity to consider trying to use JSX in the specs? Or not worth it for just this one? like we we could pass `const WrapperTag = isButtonTo ? <form action="#"> : <>` or something, right?

Reply:
I don't think JSX would be very helpful here, both because `WrapperTag` wouldn't really work out of the box as you suggest†, and because this is a custom element that has no relation to React.

I think if I was to refactor this, something like what I'd done a few lines below could help make the "wrapping" a bit more clear?

```ts
let tag = tagName === 'a' ? '<a href="#">Click Me</a>' : '<input type="submit" value="Click Me">';
if (isButtonTo) {
  tag = `<form action="#">${tag}</form>`;
}
```

Or:

```ts
let tag;
if (tagName === 'a') {
  tag ='<a href="#">Click Me</a>';
} else {
  tag = '<input type="submit" value="Click Me">';
}

if (isButtonTo) {
  tag = `<form action="#">${tag}</form>`;
}
```

†

`WrapperTag` capitalized would suggest it's the component, which would be possible to do with a reference to the tag name only, where props were handled separately.

```tsx
const WrapperTag = isButtonTo ? 'form' : Fragment;
const wrapperProps = isButtonTo ? { action: '#' } : undefined;

// ...

<div className="spinner-button__content">
  <WrapperTag {...wrapperProps}>
    {/* ...child */}
  </WrapperTag>
</div>
```

Alternatively, it could be done a bit more concisely with a `cloneElement`, closer to what you suggest:

```tsx
const wrapperElement = isButtonTo ? <form action="#" /> : <></>;

// ...


<div className="spinner-button__content">
  {cloneElement(wrapperElement, null, /* ...child */)}
</div>
```

Reply:
>this is a custom element that has no relation to React

That being said, JSX isn't necessarily limited to React, and I do know of [a library we could use](https://github.com/aduth/hijinks) to use JSX for constructing plain DOM elements!

Reply:
ok yeah this was not the rabbit hole I was expecting today 😂 

The first option seemed like the clearest, but ya definitely not a requirement to change to merge this PR
```
tag = `<form action="#">${tag}</form>`;
```